text/0000-code-literals.md

```diff
@@ -0,0 +1,417 @@
+        !```, "bar");
+
+    // Terminating newline
+    assert_eq!(```
+        bar
+        ```, "bar\n");
+
+    // Preserved indent
+    assert_eq!(```
+    if a:
+        print(42)
+    ```, "if a:\n    print(42)\n");
+
+    // Relative indent
+    assert_eq!(```
+            if a:
+                print(42)
+            ```, "if a:\n    print(42)\n");
+
+    // Relative to closing backticks
+    assert_eq!(```
+
+
+            if a:
+                print(42)
+    ```, "\n\n        if a:\n            print(42)\n");
+```
+
+The text between the opening backticks and the first newline is
+preserved within the AST, but is otherwise unused.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The main drawback is increased complexity of the language:
+
+1. It adds a new symbol to the language, which was not previously used.
+2. It adds a third way of writing string literals.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+There is lots of room to bike-shed syntax.
+If there is significant opposition to the backtick syntax, then an
+alternative syntax such as:
+```
+code"
+    string
+"
+```
+could be used.
+
+Similarly, the use of more than three backticks may be unpopular.
+It's not clear how important it is to be able to nest backticks
+within backticks, but a syntax mirroring raw string literals could
+be used instead, eg.
+```
+`# foo
+    string
+#`
+```
+
+There is also the question of whether the backtick syntax would
+interfere with the ability to paste Rust code snippets into such
+blocks. Experimentally, markdown parsers do not seem to have any
+problems with this (as demonstrated in this document).
+
+## A list of all options regarding syntax
+
+### Quote style
+
+  - :heavy_check_mark: **3+N backticks**
+    ```rust
+    let _ = ```
+        some code
+        ```;
+    ```
+
+  - **3+N double-quotes**
+    ```rust
+    let _ = """
+        some code
+        """;
+    ```
+
+  - **3+N single quotes**
+    ```rust
+    let _ = '''
+        some code
+        ''';
+    ```
+
+  - **Word prefix + N+1 hashes**
+    ```rust
+    let _ = code#"
+        some code
+        "#;
+    ```
+
+  - **Single character prefix + N+1 hashes**
+    ```rust
+    let _ = m#"
+        some code
+        "#;
+    ```
+    (note: `c` is already reserved for C strings)
+
+### Indentation rules
+
+  - :heavy_check_mark: **Relative to closing quote + retain final newline**
+
+    Benefits:
+      - Allows every possible indentation to be represented.
+      - Simple rule.
+      - The value of the string is obvious and intuitive.
+
+    Drawbacks:
+      - It is not possible to represent strings without a trailing newline.
+
+  - **Relative to closing quote + remove final newline**
+
+    Benefits:
+      - Allows every possible indentation to be represented.
+      - Simple rule.
+      - Strings without a final newline can be represented.
+
+    Drawbacks:
+      - There are two ways to represent the empty string.
```

Comment:
please list both ways, afaict the only valid way is:
````rust
let empty = ```
```;
````
since the closing triple backticks must be at the start of a line (possibly indented).

Reply:
If you want:
````rust
```
foo
``` == "foo"
````

Then you must also have:
````rust
```

``` == ""
````

Otherwise your rules are inconsistent.